src/components/users/UserRegisterForm.js

```diff
@@ -1,146 +1,108 @@
 import React, { Component } from 'react'
-import Axios from 'axios'
-import emailValidator from 'email-validator'
 
-import RaisedButton from 'material-ui/RaisedButton'
-import Snackbar from 'material-ui/Snackbar'
+import Form from '../form-elements/CustomForm'
+import TextField from '../form-elements/ValidatingTextField'
+
+import emailValidator from 'email-validator'
 
-import TextField from './ValidatingTextField'
 import { URL_USERS } from '../../constants/Constants'
+import Axios from 'axios'
+
+import Snackbar from 'material-ui/Snackbar'
 
 class UserRegisterForm extends Component {
   constructor (props) {
     super(props)
     this.state = {
       snackAutoHideDuration: 2000,
       snackMessage: 'User action',
-      snackOpen: false,
-      email: null,
-      password: null,
-      passwordRepeat: null,
-      firstName: null,
-      lastName: null,
-      validFields: {
-        email: false,
-        password: false,
-        passwordRepeat: false,
-        firstName: false,
-        lastName: false
-      }
+      snackOpen: false
     }
   }
 
-  onFieldValueChange = (fieldName, fieldValue, hasError) => {
-    this.setState(prevState => ({
-      [fieldName]: fieldValue,
-      validFields: {
-        ...prevState.validFields,
-        [fieldName]: !hasError
-      }
-    }))
-  }
-
   handleRequestClose = () => {
     this.setState({ snackOpen: false })
   }
-
-  handleSubmit = event => {
-    Axios.post(URL_USERS, {
-      email: this.state.email,
-      firstName: this.state.firstName,
-      lastName: this.state.lastName,
-      password: this.state.password
-    })
-            .then(response =>
-                this.setState({
-                  snackMessage: 'User registered',
-                  snackOpen: true
-                })
-            )
-            .catch(error =>
-                this.setState({
-                  snackMessage: 'Failed to register',
-                  snackOpen: true
-                })
-            )
-  }
-
   render () {
     return (
-      <div style={main}>
+      <div>
+        <Form
+          title='User registration'
+          buttonText='Register'
+          onSubmitCallback={data => {
+            Axios.post(URL_USERS, {
+              email: data.email,
```

Comment:
While data in `data` is same as data that you want to send, just use `Object.assign()` to copy that object.

Reply:
data have extra property passwordRepeat, so request sending copied object gets rejected statusCode: 400,
        error: 'Bad Request',
        message: 'child "body" fails because ["passwordRepeat" is not allowed]' }

Reply:
Ok, got it. In that case try  to use [LoDash](https://lodash.com/) method [`_.omit`](https://lodash.com/docs/4.17.10#omit).

Reply:
In general - get yourself comfortable with LoDash - it is use very widely and has lot's of nice features.